### PR TITLE
Update "Additional details" form in KYC wallet upgrade flow

### PR DESCRIPTION
**Pull request by marcaaron**
### Details

Improves the "Additional details" step of the KYC Wallet Upgrade flow so it behaves like the VBA.

### Fixed Issues
$ https://github.com/Expensify/Expensify/issues/188531

### Tests
1. Login with an account that has a SILVER wallet
2. Trigger the KYC flow by attempting to send money to another user via the FAB + button
3. Upload doc and video in the Onfido screen
4. Test the "Additional details" form on the next screen
5. Make sure submitting the form without any details will highlight which fields are required
6. Test that you cannot add an address with a PO Box
7. Test that server errors are presented above the "Save & continue" button and validation errors have a "Fix the errors" link
8. Close form and re-trigger. Verify that the user details are not cleared out and "remembered".

### QA Steps
Internal QA - perform tests above with a new account on the "Send Money" beta

### Tested On

- [x] Web
- [x] Mobile Web - Onfido does not work on simulator so I manually skipped that step
- [x] Desktop
- [x] iOS
- [x] Android

### CODE REVIEW

File: src/pages/EnablePayments/AdditionalDetailsStep.js
Comment (marcaaron) [inline src/pages/EnablePayments/AdditionalDetailsStep.js:168]: I thought about maybe taking the time to abstract some of the logic out of `ReimbursementAccountUtils` to reuse here, but decided not to since this will all be replaced eventually by our new forms system whenever it is ready.

https://github.com/Expensify/App/blob/9f2b4f025961d5513dcf949b896f3d0fef956a98/src/libs/ReimbursementAccountUtils.js#L32-L43

I will create a follow up to address this if it looks like the forms project will take a long time.

File: src/libs/actions/Wallet.js
Comment (nickmurray47) [inline src/libs/actions/Wallet.js:112]: +1 like the change to booleans for values, will keep this refactor in mind

File: src/pages/EnablePayments/AdditionalDetailsStep.js
Comment (nickmurray47) [inline src/pages/EnablePayments/AdditionalDetailsStep.js:256]: nice!

**Review by nickmurray47 — COMMENTED** [review]
Code looks good to me! Had a couple basic comments to add

File: src/ONYXKEYS.js
Comment (marcaaron) [reply src/ONYXKEYS.js:119]: I'm following the pattern we initially set up here:

https://github.com/Expensify/App/blob/1c80150632c684174270c019b7439854318471a4/src/ONYXKEYS.js#L133-L134

And yeah they are drafts since if the user bails on the flow and returns later they can pick up where they left off.

Comment (marcaaron) [reply src/ONYXKEYS.js:119]: This hopefully will be a detail of all forms eventually but for now we have to store it somewhere.

File: src/libs/actions/Wallet.js
Comment (marcaaron) [reply src/libs/actions/Wallet.js:86]: Sure I can get behind that reasoning.

**Comment by marcaaron** [comment]
Updated!

**Review by nickmurray47 — APPROVED** [review]
LGTM

**Comment by botify** [comment]
@nickmurray47 looks like this was merged without passing tests. Please add a note explaining why this was done and remove the `Emergency` label if this is not an emergency.

**Comment by nickmurray47** [comment]
> looks like this was merged without passing tests. Please add a note explaining why this was done and remove the `Emergency` label if this is not an emergency.

This doesn't look right to me, going to remove the `Emergency` label unless you think it's necessary @marcaaron

**Comment by marcaaron** [comment]
Ya I think this is a bug and this did pass all the tests.

**Comment by Julesssss** [comment]
I tried to verify this to unblock the [deploy checklist](https://github.com/Expensify/App/issues/6778), but unfortunately I'm blocked by [this bug](https://github.com/Expensify/App/issues/5299) in the Onfido flow for Android mobile 😞

**Comment by marcaaron** [comment]
Ah thanks @Julesssss! I think we can check this off if everything else is working - nobody should be triggering the KYC flow in product yet and we're going to need to E2E test the entire flow on all platforms eventually.

**Comment by Julesssss** [comment]
Ah that's great, thanks for confirming 👍

**Pull request merged**
